src/index.ts

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env node
+import * as program from 'commander';
+import { red, yellow, cyan } from 'colors';
+import { simulator, SimulatorConfig } from './simulator';
+const axios = require('axios');
+const cache = require('ez-cache')();
+
+const DEVICE_RAND: string = 'rand';
+
+const getConfig = (args: any, env: any): SimulatorConfig =>
+  program
+    .option(
+      '-c, --certs-response <certsResponse>',
+      'JSON returned by call to the Device API endpoint: POST /devices/{deviceid}/certificates',
+      env.CERTS_RESPONSE,
+    )
+    .option(
+      '-e, --endpoint <endpoint>',
+      'AWS IoT MQTT endpoint',
+      env.MQTT_ENDPOINT,
+    )
+    .option(
+      '-d, --device-id <deviceId>',
+      'ID of the device',
+      env.DEVICE_ID || DEVICE_RAND,
+    )
+    .option(
+      '-o, --device-ownership-code <deviceOwnershipCode>',
+      'PIN/ownership code of the device',
+      env.DEVICE_OWNERSHIP_CODE || '123456',
+    )
+    .option(
+      '-m, --mqtt-messages-prefix <mqttMessagesPrefix>',
+      'The prefix for the MQTT unique to this tenant for sending and receiving device messages',
+      env.MQTT_MESSAGES_PREFIX,
+    )
+    .option(
+      '-s, --services <services>',
+      'Comma-delimited list of services to enable. Any of: [gps,acc,temp,device]',
+    )
+    .option(
+      '-a, --app-fw-version <appFwVersion>',
+      'Version of the app firmware',
+      1,
+    )
+    .option(
+      '-k, --api-key <apiKey>',
+      'API key for nRF Cloud',
+      process.env.API_KEY,
+    )
+    .option(
+      '-h, --api-host <apiHost>',
+      'API host for nRF Cloud',
+      process.env.API_HOST || 'https://api.dev.nrfcloud.com',
+    )
+    .option('-l, --link', 'link to account', false)
+    .option('-v, --verbose', 'verbose', false)
+    .parse(args)
+    .opts() as SimulatorConfig;
+
+// i don't understand this 'no-floating-promises' rule.
+// so I'm using the javascript 'void' operator:  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/void
+// ¯\_(ツ)_/¯
+void (async (): Promise<void> => {
+  const config: SimulatorConfig = getConfig(process.argv, process.env);
+
+  const {
+    deviceId,
+    apiKey,
+    apiHost,
+    certsResponse,
+    endpoint,
+    mqttMessagesPrefix,
+    deviceOwnershipCode,
+    verbose,
+    link,
+  } = config;
+
+  const debug = (message: string) => verbose && console.log(cyan(message));
+  const info = (message: string) => console.log(yellow(message));
+  const error = (message: string) => console.log(red(message));
+
+  if (deviceId === DEVICE_RAND) {
```

Comment:
Why not remove use of `DEVICE_RAND` and just do `if (!deviceId)...`?

Reply:
👍